configure.ml

```diff
@@ -1079,6 +1079,31 @@ let write_dbg_wrapper f =
 
 let _ = write_dbg_wrapper "dev/ocamldebug-coq"
 
+(** * Generate an implementation of 63-bit arithmetic *)
+
+let ml_file_copy input output =
+  let i = open_in input in
+  let o = open_out output in
+  let pr s = fprintf o s in
+  pr "(* DO NOT EDIT THIS FILE: automatically generated by ../configure *)\n";
+  pr "(* see kernel/uint63_amd64.ml and kernel/uint63_x86.ml *)\n";
+  try
+    while true do
+      output_string o (input_line i); output_char o '\n'
+    done
+  with End_of_file ->
+    close_in i;
+    close_out o;
+    Unix.chmod output 0o555
+
+let write_uint63 () =
+  if max_int = 1073741823 (* 32-bits *) then
+    ml_file_copy "kernel/uint63_x86.ml" "kernel/uint63.ml"
+  else (* 64 bits *)
+    ml_file_copy "kernel/uint63_amd64.ml" "kernel/uint63.ml"
```

Comment:
Why not use the same system as is already there to share names/esubst in the kernel and checker?

Reply:
I think this is not the right place to generate these files [neither for the checker stuff]. The files should be generated from the Makefiles so we have proper dependency tracking.

Reply:
How do you suggest to test the size of OCaml's `int`s from the `Makefile`?

Reply:
We discussed this in the chat and we thought that a sensible mechanism could be:
- have `configure` generate a make variable `OCAML_ARCH` obtained by parsing the homonymous  `ocamlc -config` field.
- add a rule to `Makefile.build` of the form:
  ```
  kernel/uint63.ml: kernel/uint63_$(ARCH).ml.in
      cp -a $< $@
  ```
- If more tweaking is needed, `cpp` could be used in the standard way.